pkg/webhook/build.go

```diff
@@ -53,23 +53,29 @@ func (ac *AdmissionController) validateBuild(ctx context.Context, _ *[]jsonpatch
 	// If a build specifies a template, all the template's parameters without
 	// defaults must be satisfied by the build's parameters.
 	var volumes []corev1.Volume
-	var tmpl *v1alpha1.BuildTemplate
+	var tmpl v1alpha1.BuildTemplateInterface
 	if b.Spec.Template != nil {
 		tmplName := b.Spec.Template.Name
 		if tmplName == "" {
 			return validationError("MissingTemplateName", "the build specifies a template without a name")
 		}
 
 		// Look up the template in the Build's namespace.
-		tmpl, err = ac.buildClient.BuildV1alpha1().BuildTemplates(b.Namespace).Get(tmplName, metav1.GetOptions{})
-		if err != nil {
-			return err
+		if b.Spec.Template.Kind == v1alpha1.ClusterBuildTemplateKind {
+			tmpl, err = ac.buildClient.BuildV1alpha1().ClusterBuildTemplates().Get(tmplName, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+		} else {
```

Comment:
We should reject a build that specifies `kind: Nonsense` instead of assuming anything besides `ClusterBuildTemplate` means `BuildTemplate`.

Should we reject builds where `kind` is not specified? Or assume they mean one or the other?

Reply:
added a check to make sure the kind is either `BuildTemplate` or `ClusterBuildTemplate` and assumed `BuildTemplate` as the default if nothing is specified. Will also throw an error if other values are used.